src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -134,6 +134,22 @@ public boolean hasFolder(Folder folder) {
         return folders.contains(folder);
     }
 
+    /**
+     * Checks and returns true if folder already exists
+     */
+    public boolean hasFolderName(FolderName folderName) {
+        requireNonNull(folderName);
+        return folders.containsFolderName(folderName);
+    }
```

Comment:
In this class, there is also a method `hasFolder()` that checks if a Folder of the same identity exists, so you do not have to write another one.